tests/tests_provider.py

```diff
@@ -0,0 +1,76 @@
+from nameko.rpc import rpc
+from nameko.testing.services import worker_factory
+
+from nameko_vault import VaultProvider
+
+
+class TestVaultService:
+    name = "test_vault_service"
+    vault = VaultProvider(mount_point="test")
+
+    @rpc
+    def create_secret(self, path, secret):
+        response = self.vault.create_or_update_kv_secret(path, secret)
+
+        return response
+
+    @rpc
+    def patch_secret(self, path, secret):
+        response = self.vault.patch_kv_secret(path, secret)
+
+        return response
+
+
+def test_vault_service_create_secret():
+    service = worker_factory(TestVaultService)
+
+    path = "path/test"
+    secret = {"example": "Test", "number": 42}
+    mock_return = {
+        'request_id': '4ce62ee7-0f88-3efc-d745-5e2fbc423789',
+        'lease_id': '',
+        'renewable': False,
+        'lease_duration': 0,
+        'data': {
+            'created_time': '2020-09-10T00:25:40.92411625Z',
+            'deletion_time': '',
+            'destroyed': False,
+            'version': 1,
+        },
+        'wrap_info': None,
+        'warnings': None,
+        'auth': None,
+    }
+
+    service.vault.create_or_update_kv_secret.return_value = mock_return
+
+    assert service.create_secret(path, secret) == mock_return
+    service.vault.create_or_update_kv_secret.assert_called_once_with(
+        path, secret
+    )
```

Comment:
I think those tests are testing the example service behavior, not the `VaultProvider` behavior.

Note: this can also be addressed in another PR, the CI pipeline with tests running is valuable

Reply:
OK this one I will leave for a future PR